### ISSUE TITLE
NFS mount fails on subsequent `vagrant up` commands

### ISSUE THREAD

**Issue by n1vlac** [hashicorp/vagrant#1941]
I know there are other issues around this problem, but perhaps my situation is a little different.

So I'm hitting errors on mounting a shared directory. The funny thing is, it WORKS when I create a new VM, though it seems to hit a bunch of errors before somehow finally working. See below:

```
DEBUG ssh: Exit status: 0
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: mount.nfs: requested NFS version or transport protocol is not supported

DEBUG ssh: Exit status: 32
 INFO retryable: Retryable exception raised: #<Vagrant::Errors::LinuxNFSMountFailed: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work>
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: mount.nfs: requested NFS version or transport protocol is not supported

DEBUG ssh: Exit status: 32
 INFO retryable: Retryable exception raised: #<Vagrant::Errors::LinuxNFSMountFailed: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work>
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: mount.nfs: requested NFS version or transport protocol is not supported

DEBUG ssh: Exit status: 32
 INFO retryable: Retryable exception raised: #<Vagrant::Errors::LinuxNFSMountFailed: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work>
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: Exit status: 0
DEBUG ssh: Checking whether SSH is ready...
DEBUG ssh: Re-using SSH connection.
 INFO ssh: SSH is ready!
 INFO guest: Execute capability: shell_expand_guest_path (ubuntu)
```

But once it's created, on any subsequent `vagrant up` commands (after I do a `halt`), it just fails.

```
mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work>
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: mount.nfs: requested NFS version or transport protocol is not supported

DEBUG ssh: Exit status: 32
 INFO retryable: Retryable exception raised: #<Vagrant::Errors::LinuxNFSMountFailed: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work>
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: mount.nfs: access denied by server while mounting 10.98.0.1:/Users/nivlac/work/nucleus

The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3 10.98.0.1:'/Users/nivlac/work/nucleus' /home/vagrant/work
```

I've been using Vagrant for several months now... this error just started happening yesterday. 

Vagrant version 1.2.4

**Comment by epsiloneridani** [comment]
@mogetutu  Thanks, I had exactly the same on Ubuntu 14.04, I switched off firewall and installation went well, everything works now, so the question is how to configure firewall properly for this issue ?

**Comment by benjifisher** [comment]
@shishirsharma :  Your solution worked for me. I halted all running VM's, ran "sudo ifconfig ... down" for all the vboxnet networks, and "vagrant up" worked. Previously, I rebooted my host machine, but I did not like that "solution".

**Comment by fnordfish** [comment]
Just experienced the same ('mount.nfs: requested NFS version or transport protocol is not supported') on OS X Yosemite 10.10.2, vagrant 1.7.2, VirtualBox 4.3.20
After disabling UDP for the nfs connection it worked again: `config.vm.synced_folder ".", "/vagrant", type: "nfs", nfs_udp: false`
Not sure whats the NFS-UDP status on OS X is right now.

**Comment by phpguru** [comment]
For what it's worth

> Quote @fnordfish Just experienced the same ('mount.nfs: requested NFS version or transport 
> protocol is not supported') on OS X Yosemite 10.10.2, vagrant 1.7.2, VirtualBox 4.3.20

Same exact configuration - works on my box, doesn't work on my colleague's box. We cannot seem to figure out any differences between our two MacbookPro's after 8 hours of scouring each other's configs, caches, and googling the errors, uninstalling and reinstalling everything. To no avail. He gets asked for docker@127.0.0.1's password and I don't (it's tcuser). NFS mounts work fine on my box but not on his. Even ran disk utility to see if there were bad folder permissions. It's truly mind-boggling.

**Comment by crobays** [comment]
Never thought this would be the problem but i fixed it by setting my `/etc/hosts` to the default.
Thanks @therobyouknow !
Hope somebody else is helped by this

**Comment by ahzab** [comment]
Thanks @shishirsharma  Your solution worked for me 

>$ sudo ifconfig vboxnet0 down
>$ sudo ifconfig vboxnet0 up

**Comment by lambertb** [comment]
For me the problems was also caused by the /etc/hosts file of my OSX: I have two vagrant boxes, on different ip-addresses (192.168.56.x and 192.168.55.x). Initially, the both worked, but after restart of my system, I had disable one of these ip-addresses in my hosts file to run a 'vagrant up'.

Can someone explain what is the relation between the hosts file and the mounting of external paths by NFS?

**Comment by tristanbes** [comment]
I'm facing the same problem: 
`Vagrant 1.7.2`
`VirtualBox 4.3.28`

`OSX 10.10.3`

Plugins:
`vagrant-cachier (1.2.0)`
`vagrant-hostmanager (1.5.0)`
`vagrant-share (1.1.3, system)`

I tried on a macbook air 10.10.3 and it worked with the same settings/versions that on my macbook pro that doesn't. I can't figure out why it's happening to me :/

**Comment by jeremy-bruns** [comment]
installing these
 ```sudo apt-get install nfs-common nfs-kernel-server``` 
fixed it for me.

**Comment by rceee** [comment]
For me it was a malformed PATH variable.  Fixing it in the bash config files, then exiting and restarting the shell completely (sourcing will just append) fixed it.

**Comment by fireproofsocks** [comment]
Vagrant used to work fine... now "vagrant up" is prompting for a sudo password -- I suspect that might be related to HFS timing out.  Concerned that the user on the host system needs sudo...

**Comment by yellowmamba** [comment]
I am having the same issue with @tristanbes . I have `sudo apt-get install nfs-common nfs-kernel-server` run in my guest ubuntu machine, but `vagrant reload` still spits `mount.nfs: requested NFS version or transport protocol is not supported`.

**Comment by dagonzalezc** [comment]
i have a Mac and had the same problem, the solution for me was changing all the synced_folder in Vagrantfile to nfs_udp:false and nfs: true, everything works in TCP protocol and the default for vagrant is UDP.

**Comment by bartmcleod** [comment]
Same problem on a Mac since Monday december 14, 2015 after a system reboot. I thought it had to do with installing VMware 8.0.2, but restoring 8.0.0 did not fix the issue. The issue appears with all my VM's. None of the suggestions listed here helped so far. I also tried to issue the command manually from within the VM, also with tcp, but it did not help. Any mismatch between nfs on the guest and nfs on the host (such as a wrong /etc/hosts file) can trigger the error. The same error could come from a port mismatch for nfs (default is 2049). Just in my case, I wasn't lucky enough to find the cause and I am stuck with machines that do not properly sync their files, because I have to turn off nfs to be able to share any files at all. I re-installed Vagrant, I updated the plugins, I reinstalled VMware, I destroyed and rebuilt the boxes, I installed the nfs server inside the box, nothing helped.

**Comment by redrohX** [comment]
@c-castillo solution works for me. :+1:

**Comment by tibotiber** [comment]
For me it's similar to @lambertb: with 2 VMs, I had to remove the mapping of the VMs' IP to localhost from `/etc/hosts` (OS X Yosemite).

**Comment by wieczorek1990** [comment]
I just had to:
```sh
# service nfs-kernel-server restart
```

**Comment by sagotsky** [comment]
Installing nfs-common and nfs-kernel-service wasn't enough for me.  I had to start up both services in debian.

**Comment by mbifulco** [comment]
Just ran into this issue myself, with macOS Sierra as the host OS.  Restarting the host seems to have done the trick to fix it for me, but I haven't the slightest clue why.

**Comment by fengler-it** [comment]
Had nfs timoeout while trying to mount from guest to ubuntu 16.04.01 host.
After disabling ufw, it worked! :-)

**Comment by gkatsanos** [comment]
https://github.com/mitchellh/vagrant/issues/1941#issuecomment-43098597 fixed it for me. adding `127.0.0.1` to the hosts file and 
```
$ sudo ifconfig vboxnet0 down
$ sudo ifconfig vboxnet0 up
```

**Comment by nicklaw5** [comment]
@fengler-it's solution worked for me:

```
$ sudo ufw disable
$ vagrant up --provision
$ sudo ufw enable
```

**Comment by tecnocat** [comment]
For Mac OS Sierra this solve my day! (Vagrant 1.9.1)

https://github.com/mitchellh/vagrant/issues/5424#issuecomment-284816293

**Comment by chadfurman** [comment]
I had a similar problem in Ubuntu, Vagrant version 1.9.3, VB version 5.1.22

The problem was that my host's OS has changed the `ifconfig` to the `ip` command, and `vboxnet0` did not have an ip address. 

I had to run:

`ip a add 192.168.33.1/255.255.255.0 dev vboxnet0` before Vagrant could start the NFS share

There were many other things I tried: starting/stopping the firewall (ufw, iptables, both, neither); adding specific rules to the firewall to allow all traffic to this IP; `service start` for each of nfs-common, nfs-kernel-server, nfs-server, nfs-client; and clearing out my /etc/exports

It wasn't until I made sure my vboxnet0 had an IP on the host machine that everything worked.

**Comment by leonerd** [comment]
For another take on this: in current debian it seems that `nfs-kernel-server` is not sufficient for vagrant any more. I had to remove that in favour of the userland implementation of `nfs-server`, which made it run happily.

```
$ vagrant --version
Vagrant 1.9.8
```

**Comment by whalesingswee** [comment]
Would like to add if you're using OSX 10.14 and running this issue, please follow this simple step would help https://github.com/hashicorp/vagrant/issues/10234

> The Mojave upgrade removes/disables NFS, resulting the /etc/exports being renamed to  #/etc/exportsbak

**Comment by davthu** [comment]
I updated Vagrant to the latest version and it solved it for me (macOS 10.13.6)

**Comment by saggzz** [comment]
Had the same issue after i tried to upgrade in virtualbox vagrant (stupid). I tried almost everything from above and pretty much from any other forum out there. I the end i got it working half way, but the project files from the vagrant project were not piped in the new environment. What worked in the end:
1. delete Virtualbox including `~/'VirtualBox`' and `~./.config/VirtualBox`.
2. delete vagrant in `/opt/` and `/usr/bin  `
3. delete in `/etc/hosts` and `/etc/exports` all listings which are dependent to your virtualbox/vagrant environment.
4. Install it all again 
5. If possible git clone your project form an earlier stage to a new dir.
6. copy your changes from the old files stepwise to the new project. 
Its kinda rollback, but it was the only way i got it working.
